#! /bin/sh /usr/share/dpatch/dpatch-run
## quiesce-bonjour-warning.dpatch by  <till.kamppeter@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Silence Avahi Bonjour compat warning, since this can cause SIGPIPE
## DP: crashes if stderr is not available.
## DP: http://bugs.debian.org/497492, http://www.cups.org/str.php?L2931

@DPATCH@
diff -urNad cups-1.4.0~svn8773~/scheduler/printers.c cups-1.4.0~svn8773/scheduler/printers.c
--- cups-1.4.0~svn8773~/scheduler/printers.c	2009-08-23 12:16:58.000000000 +0200
+++ cups-1.4.0~svn8773/scheduler/printers.c	2009-08-23 17:52:46.000000000 +0200
@@ -4397,6 +4397,9 @@
     }
 
 #ifdef HAVE_DNSSD
+    /* silence Avahi Bonjour compat warning, since this can cause SIGPIPE
+     * crashes if stderr is not available */
+    setenv("AVAHI_COMPAT_NOWARN", "1", 0);
     cupsdSetString(&p->product, ppd->product);
 #endif /* HAVE_DNSSD */
 
